woocommerce/woocommerce
# Maybe default the selected state as the first one when it's "*"

### Changes proposed in this Pull Request:

When WC is provided only the country for the option `woocommerce_default_country` then it will fail to select a default state within the listed states.

Try this by manually setting `woocommerce_default_country` to "US" and you will see it selects the first country in the list instead of the first state within that country.

This only happens for countries that have states like US and CA for instance.

If the default state is intended to be "*" which is getting passed to this function, then that logic should pass along to countries with states too so that it will always default to the first one. Otherwise, the country never gets selected and that flow is broken.



### How to test the changes in this Pull Request:

1. Create a WC store (but don't complete onboarding)
2. Manually set option: `wp option update woocommerce_default_country US` (use US or CA or any other country that has states that are selectable)
3. Go to the WooCommerce > Settings page
4. Look at the "Country / State" option, it should be now default to the first state for that country (previously it would set as "Afghanistan")
